prov/efa/src/efa_mr.c

```diff
@@ -57,6 +57,62 @@ static struct fi_ops efa_mr_cache_ops = {
 	.ops_open = fi_no_ops_open,
 };
 
+/*
+ * @brief Validate HMEM attributes and populate efa_mr struct
+ *
+ * Check if FI_HMEM is enabled for the domain, validate whether the specific
+ * device type requested is currently supported by the provider, and update the
+ * efa_mr structure based on the attributes requested by the user.
+ *
+ * @params[in]	efa_mr	efa_mr structure to be updated
+ * @params[in]	attr	fi_mr_attr from the user's registration call
+ *
+ * @return FI_SUCCESS or negative FI error code
+ */
+static int efa_ep_mr_hmem_setup(struct efa_mr *efa_mr,
+                                const struct fi_mr_attr *attr)
+{
+	/*
+	 * Host memory is always supported, but is not set in
+	 * supported_hmem_devs, so check this before continuing.
+	 */
+	if (attr->iface == FI_HMEM_SYSTEM) {
+		efa_mr->peer.iface = FI_HMEM_SYSTEM;
+		return FI_SUCCESS;
+	} else if (efa_mr->domain->util_domain.info_domain_caps & FI_HMEM) {
+		/*
+		 * Skipping the domain type check above is okay here since
+		 * util_domain is at the beginning of both efa_domain and
+		 * rxr_domain.
+		 */
+		if (efa_mr->domain->ctx->supported_hmem_devs & attr->iface) {
```

Comment:
I do not think we can use `attr->face` as flag, because it is an enum.

For example, FI_HMEM_ZE's value is 3, which is FI_HMEM_CUDA |  FI_HMEM_ROCR.

Also, can we just call `ofi_hmem_is_initialized(attr->iface)` here?

Reply:
Yep, you're right, I'll fix this.

I think we still need internal state in our provider to decide whether it supports that device type. That device type could be supported by Libfabric, but not the provider.

Reply:
I thought about it some more and removed the flag for now. We'll want to store additional information, like if p2p is supported/enabled, and it's more appropriate to store that in the endpoint.

For now `domain_caps & FI_HMEM` validates what we need and we can extend this later.